src/i18nDefinitionProvider.ts

```diff
@@ -35,72 +25,24 @@ export class I18nDefinitionProvider implements DefinitionProvider {
             return null;
         }
 
-        return this.findKeyValueLocationInDocument(translationPart.file, key, locale);
+        return this.findKeyValueLocationInDocument(translationPart.file, key, locale, translationPart.yamlDocument);
     }
 
-    findKeyValueLocationInDocument(file: Uri, absoluteKey: string, locale: string): Thenable<Location> {
+    findKeyValueLocationInDocument(file: Uri, absoluteKey: string, locale: string, yamlDocument?: YAMLDocument): Thenable<Location> {
```

Comment:
I started persisting this yamlDocument stuff since it's an intermediate state of fully parsing the doc and we need to run it at least once on each translation file anyway. Then we don't have to necessarily incur the overhead of running the parser every time.